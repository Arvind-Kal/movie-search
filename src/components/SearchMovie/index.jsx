import React, {useState} from 'react';
import styles from './styles/SearchMovie.module.scss';
import MovieCard from '../MovieCard';

const SearchMovie = () => {
    const [query, setQuery] = useState('');
    const [movies, setMovies] = useState([]);

    const searchMovie = async (e) => {
        e.preventDefault();

        if (query) {
            const url = `https://api.themoviedb.org/3/search/movie?api_key=421c89b0d7ac0a1e5d5daa5067038be9&language=en-US&query=${query}&page=1&include_adult=false`;

            try {
                const res = await fetch(url);
                const data  = await res.json();
                setMovies(data.results);
            } catch(err) {
                console.error(err);
            }
        }
    };

    return (
        <div className='movieContainer'>
            <form className={styles.form} onSubmit={searchMovie}>
                <label className={styles.label} htmlFor='query'>Movie's Name</label>
                <input 
                    className={styles.input} 
                    type='text' 
                    name='query' 
                    placeholder='i.e. Star Wars'
                    value={query}
                    onChange={(e) => {setQuery(e.target.value)}}></input>
                <button className={styles.button}>Search</button>
            </form>
            <MovieCard movies={movies}></MovieCard>
        </div>
    );
};

export default SearchMovie;
