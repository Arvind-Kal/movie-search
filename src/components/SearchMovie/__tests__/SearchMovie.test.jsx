import { render, waitFor, screen } from '@testing-library/react';
import SearchMovie from '../lib';

describe('SearchMovie', () => {
    it('Should render the form', async () => {
        const { asFragment } = render(<SearchMovie></SearchMovie>);
        expect(asFragment()).toMatchSnapshot();
        const searchMovieContainer = await waitFor(() => screen.findByTestId('SearchMovie__Container'));
        expect(searchMovieContainer).toBeInTheDocument();
        expect(searchMovieContainer).toMatchSnapshot();
    });
});