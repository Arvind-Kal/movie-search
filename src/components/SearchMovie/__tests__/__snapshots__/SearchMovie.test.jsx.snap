// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchMovie Should render the form 1`] = `
<DocumentFragment>
  <div
    class="movieContainer"
    data-testid="SearchMovie__Container"
  >
    <form
      class="form"
    >
      <label
        class="label"
        for="query"
      >
        Movie's Name
      </label>
      <input
        class="input"
        name="query"
        placeholder="i.e. Star Wars"
        type="text"
        value=""
      />
      <button
        class="button"
      >
        Search
      </button>
    </form>
    <div
      class="cardList"
    />
  </div>
</DocumentFragment>
`;

exports[`SearchMovie Should render the form 2`] = `
<div
  class="movieContainer"
  data-testid="SearchMovie__Container"
>
  <form
    class="form"
  >
    <label
      class="label"
      for="query"
    >
      Movie's Name
    </label>
    <input
      class="input"
      name="query"
      placeholder="i.e. Star Wars"
      type="text"
      value=""
    />
    <button
      class="button"
    >
      Search
    </button>
  </form>
  <div
    class="cardList"
  />
</div>
`;
